docs/docs/add-seo-component.md

```diff
@@ -0,0 +1,199 @@
+      site: {
+        siteMetadata: {
+          defaultTitle,
+          titleTemplate,
+          defaultDescription,
+          siteUrl,
+          defaultImage,
+          twitterUsername,
+        }
+      }
+    }) => {
+      const seo = {
+        title: title || defaultTitle,
+        description: description || defaultDescription,
+        image: `${siteUrl}${image || defaultImage}`,
+        url: `${siteUrl}${pathname || '/'}`,
+      }
+
+      return ()
+    }}
+  />
+)
+
+export default SEO
+```
+
+The last step is to return this data with the help of `Helmet`. Your complete SEO component should look like:
+
+```jsx:title=src/components/SEO.js
+import React from "react"
+import Helmet from "react-helmet"
+import PropTypes from "prop-types"
+import { StaticQuery } from "gatsby"
+
+const SEO = ({ title, description, image, pathname, article }) => (
+  <StaticQuery
+    query={query}
+    render={({
+      site: {
+        siteMetadata: {
+          defaultTitle,
+          titleTemplate,
+          defaultDescription,
+          siteUrl,
+          defaultImage,
+          twitterUsername,
+        }
+      }
+    }) => {
+      const seo = {
+        title: title || defaultTitle,
+        description: description || defaultDescription,
+        image: `${siteUrl}${image || defaultImage}`,
+        url: `${siteUrl}${pathname || '/'}`,
+      }
+
+      return (
+        <>
+          <Helmet>
+            <meta name="description" content={seo.description} />
+            <meta name="image" content={seo.image} />
+            {seo.url && <meta property="og:url" content={seo.url} />}
+            {(article ? true : null) && <meta property="og:type" content="article" />}
+            {seo.title && <meta property="og:title" content={seo.title} />}
+            {seo.description && <meta property="og:description" content={seo.description} />}
+            {seo.image && <meta property="og:image" content={seo.image} />}
+            <meta name="twitter:card" content="summary_large_image" />
+            {twitterUsername && <meta name="twitter:creator" content={twitterUsername} />}
+            {seo.title && <meta name="twitter:title" content={seo.title} />}
+            {seo.description && <meta name="twitter:description" content={seo.description} />}
+            {seo.image && <meta name="twitter:image" content={seo.image} />}
+          </Helmet>
+        </>
+      )
+    }}
+  />
+)
+
+export default SEO
+
+SEO.propTypes = {
+  title: PropTypes.string,
+  description: PropTypes.string,
+  image: PropTypes.string,
+  pathname: PropTypes.string,
+  article: PropTypes.bool,
+}
+
+SEO.defaultProps = {
+  title: null,
+  description: null,
+  image: null,
+  pathname: null,
+  article: false,
+}
+```
+
+## Examples
+
+You could also put the Facebook and Twitter meta-tags into their own components, add custom favicons you placed in your `static` folder, and add JSONLD data. To see how that works you can have a look at these two examples:
```

Comment:
Probably worth linking to more info on JSON-LD here, since most people probably don't know what that is.

It also might be better to talk about schema.org instead of JSON-LD, since schema.org _uses_ JSON-LD to read SEO data, but JSON-LD is a spec that can be used for things other than SEO.